content/en/blog/_posts/Kubernetes-setup-using-Ansible-and-Vagrant.md

```diff
@@ -0,0 +1,248 @@
+---
+layout: blog
+title: Kubernetes setup using Ansible and Vagrant
+date: 2019-02-06
+---
+
+## Objective
+This blog describes the steps required to setup multi node Kubernetes cluster for development purpose. This setup provides production like cluster that can be setup in local machine.  
+
+## Why do we require multi node cluster setup?
+Multi node Kubernetes cluster offers production like environment which in turn has various advantages. Even though Minikube provides an excellent platform for getting started it doesn't provide the opportunity to work with multi node cluster which can help solve problems or bugs that are related to application design and architecture. For instance, Ops can reproduce an issue in multi node cluster environment, Testers can deploy multiple version of application for executing test cases and verifying changes. These benefits enable teams to resolve issues faster which can contribute to being agile. 
+
+## Why use Vagrant and Ansible?
+Vagrant is a tool that will allow us to create a virtual environment easily and it eliminates pitfalls that cause the works-on-my-machine phenomenon. It can be used with multiple providers such as Oracle VirtualBox, VMware, Docker and so on. It allows us to create disposable environment by making use of configuration files. 
+
+Ansible is Infrastructure automation engine that automates software configuration management. It is agent less and allows us to use ssh keys for connecting to remote machines. Ansible playbooks are written in yaml and it offers inventory management in simple text files.
+
+
+### Pre-requisites
+- Vagrant should be installed on your machine. Installation binaries can be found [here](https://www.vagrantup.com/downloads.html).
+- Oracle VirtualBox can be used as Vagrant provider or make use of similar providers as described in official [documentation](https://www.vagrantup.com/docs/providers/) of Vagrant.
+- Ansible should be installed in your machine. Refer Ansible installation [guide](https://docs.ansible.com/ansible/latest/installation_guide/intro_installation.html) for platform specific installation.
+
+## Setup overview
+We will be setting up a Kubernetes cluster that will consist of one master and two worker nodes. All the nodes will run Ubuntu Xenial 64-bit OS and Ansible playbooks will be used for provisioning.
+
+#### Step 1: Creating a Vagrantfile
+Use text editor of your choice and create a file with name Vagrantfile and insert the below code. The value of N denotes the no. of nodes present in the cluster, it can be modified accordingly. In the below example we are setting the value of N as 2.
```

Comment:
```suggestion
Use the text editor of your choice and create a file with named `Vagrantfile`, inserting the code below. The value of N denotes the number of nodes present in the cluster, it can be modified accordingly. In the below example we are setting the value of N as 2.
```